File: 0059-spiral-matrix-ii/0059-spiral-matrix-ii.py
class Solution:
    def generateMatrix(self, n: int) -> List[List[int]]:
        top,bottom ,left ,right = 0,n-1,0,n-1
        c,t = 1,n*n
        res = [[0] * n for _ in range(n)]        
        while(c<=t):
            for i in range(left,right+1):
                if c > t:
                  break
                res[top][i] = c
                c+=1
            
            top+=1
            for i in range(top,bottom+1):
                if c > t:
                  break
                res[i][right] = c;
                c+=1
            
            right-=1
            for i in range(right,left-1,-1):
                if c > t:
                  break
                res[bottom][i] = c;
                c+=1
            
            bottom-=1
            for i in range(bottom,top-1,-1):
                if c > t:
                  break
                res[i][left] = c
                c+=1
            
            left+=1
        
        return res
        

File: 0059-spiral-matrix-ii/0059-spiral-matrix-ii.cpp
class Solution {
public:
    vector<vector<int>> generateMatrix(int n) {
        int top =0 , bottom = n-1,left =0,right = n-1;
        int c =1 , t = n*n;
        vector<vector<int>>res(n,vector<int>(n,0));
        while(c<=t){
            for(int i=left;i<=right && c<=t;i++){
                res[top][i] = c;
                c++;
            }
            top++;
            for(int i=top;i<=bottom && c<=t;i++){
                res[i][right] = c;
                c++;
            }
            right--;
            for(int i=right;i>=left && c<=t;i--){
                res[bottom][i] = c;
                c++;
            }
            bottom--;
            for(int i=bottom;i>=top && c<=t;i--){
                res[i][left] = c;
                c++;
            }
            left++;
        }
        return res;

    }
};